### PR TITLE
Improve ndarray initializing performance

### PR DESCRIPTION
**Pull request by okuta**
rel #1340.

### CODE REVIEW

File: cupy/core/_dtype.pyx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+cdef dict _dtype_dict
+
+
+cdef _init_dtype_dict():
+    global _dtype_dict
+    _dtype_dict = {i: numpy.dtype(i)
+                   for i in six.integer_types + (float, bool, complex, None)}
+    for i in 'dfDFeqlihbQLIHB?':
```
Comment (toslunar) [inline cupy/core/_dtype.pyx]: Could you explain reason of this order?  Is it good idea to add strings of all supported types, like `typecodes` in https://github.com/numpy/numpy/blob/master/numpy/core/numerictypes.py?

**Comment by toslunar** [comment]
Why don't you remove the line `cdef str _all_type_chars = 'dfDFeqlihbQLIHB?'` from `elementwise.pxi`

**Comment by okuta** [comment]
I fixed.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**